.github/workflows/test.yml

```diff
@@ -0,0 +1,83 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Test
+
+on:
+  pull_request:
+    branches:
+    - main
+  push:
+    branches:
+    - main
+  schedule:
+    # GitHub removes infrequently accessed caches. Run once a week to keep the cache from going stale.
+    - cron: '0 0 * * 1'
+
+jobs:
+  test:
+    name: bazel test (${{ matrix.name }})
+
+    runs-on: ubuntu-22.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - name: 'opt'
+          flags: -c opt --define=crypto=system
+        - name: 'ASan'
+          flags: --config=clang-asan
+        - name: 'TSan'
+          flags: --config=clang-tsan
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: bazel cache
+      uses: actions/cache/restore@v3
+      with:
+        path: /tmp/bazel/cache
+        key: bazel-cache-${{ matrix.name }}-${{ github.head_ref || github.ref_name }}-${{ hashFiles('WORKSPACE', '.bazelrc', '.bazelversion', 'bazel/dependencies.bzl', 'bazel/repositories.bzl') }}
```

Comment:
Add a comment to explain this line?

Can we to avoid duplication with save action? Don't spend more than 5 minutes.

Reply:
Done.